scaleway/resource_redis_cluster_test.go

```diff
@@ -298,6 +308,76 @@ func TestAccScalewayRedisCluster_Settings(t *testing.T) {
 	})
 }
 
+func TestAccScalewayRedisCluster_Endpoints(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayRedisClusterDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_redis_cluster" "main" {
+						name = "test_redis_endpoints"
+    					version = "6.2.6"
+    					node_type = "MDB-BETA-M"
+    					user_name = "my_initial_user"
+    					password = "thiZ_is_v&ry_s3cret"
```

Comment:
I only wrote a single test for now but ideally i would have liked to test the following steps : 

1. without anything specified, so that it creates a public network
2. then with 1 private network, so that it sets a new endpoint and deletes the public network
3. then with 2 private networks, so that it adds one more
4. then change one of them, to check if the old one is deleted
5. and finally specify a public network and check that the private networks are deleted and that the public network is working.

Is this overkill ?

Reply:
It is a great plan :)